src/tfr_executive/src/teleop_action_server.cpp

```diff
@@ -0,0 +1,173 @@
+         * - Result: none
+         * */
+        void processCommand(const tfr_msgs::TeleopGoalConstPtr& goal)
+        {
+            geometry_msgs::Twist move_cmd{};
+            auto code = static_cast<tfr_utilities::TeleopCode>(goal->code);
+            if (code == tfr_utilities::TeleopCode::NONE)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, RESET");
+                //all zeros by default
+                drivebase.publish(move_cmd);
+            }
+            else if(code == tfr_utilities::TeleopCode::FORWARD)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, FORWARD");
+                move_cmd.linear.x = drive_stats.getLinear();
+                drivebase.publish(move_cmd);
+            }
+            else if (code == tfr_utilities::TeleopCode::BACKWARD)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, BACKWARD");
+                move_cmd.linear.x = -drive_stats.getLinear();
+                drivebase.publish(move_cmd);
+            }
+            else if (code == tfr_utilities::TeleopCode::LEFT)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, LEFT");
+                move_cmd.angular.z = drive_stats.getAngular();
+                drivebase.publish(move_cmd);
+           }
+            else if (code == tfr_utilities::TeleopCode::RIGHT)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, RIGHT");
+                move_cmd.angular.z = -drive_stats.getAngular();
+                drivebase.publish(move_cmd);
+            }
+            else if (code == tfr_utilities::TeleopCode::DIG)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, DIG");
+                tfr_msgs::DurationSrv digging_time;
+                ros::service::call("digging_time", digging_time);
+                ROS_INFO("Autonomous Action Server: digging time retreived %f",
+                    digging_time.response.duration.toSec());
+                //TODO hook up digging when ready
+            }
+            else if (code == tfr_utilities::TeleopCode::DUMP)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, DUMP");
+                //TODO hook up dumping when ready
+            }
+            else if (code == tfr_utilities::TeleopCode::RESET)
+            {
+                ROS_INFO("Teleop Action Server: Command Revieved, RESET");
+                //TODO hook up reset when ready
+            }
+            else
+            {
+                ROS_WARN("Teleop Action Server: UNRECOGNIZED COMMAND");
+            }
+
+            tfr_msgs::TeleopResult result{};
+            server.setSucceeded(result);
+        }
+
+        actionlib::SimpleActionServer<tfr_msgs::TeleopAction> server;
+        ros::Publisher drivebase;
+        DriveVelocity &drive_stats;
+};
+
+
+int main(int argc, char** argv)
+{
+    ros::init(argc, argv, "teleopaction_server");
```

Comment:
Personal opinion, change this to "teleop_action_server"?

Reply:
This is a typo, yup